packages/keys/src/functions/index.ts

```diff
@@ -13,9 +18,39 @@ const hex = {
   },
 };
 
-const sha256 = (data: string): string => createHash('sha256').update(data).digest('hex');
+export const sha256 = (data: string): string => createHash('sha256').update(data).digest('hex');
+
+const algToSecCrv = (alg: Algorithms) => {
+  if (alg === Algorithms.Secp256k1) {
+    return 'P-256K';
+  } if (alg === Algorithms.Secp256r1) {
+    return 'P-256';
+  }
+  throw new Error(`Algorithm ${alg} is not supported`);
+};
+/**
+ *
+ * @param key compressed or uncompressed public key or private key in hexadecimal format
+ * @param alg
+ * @returns
+ */
+export const pubToPem = (key: string, alg: Algorithms) => {
+  if (!key.startsWith('0x')) {
+    key = `0x${key}`;
+  }
+  const uncompressedPubKey = computePublicKey(key, false);
+  return new ECKey({
+    crv: algToSecCrv(alg),
+    publicKey: Buffer.from(uncompressedPubKey.slice(2), 'hex'),
+  }).toString('pem');
+};
 
-export {
-  hex,
-  sha256,
+export const privToPem = (key: string, alg: Algorithms) => {
+  if (key.startsWith('0x')) {
+    key = key.slice(2);
+  }
+  return new ECKey({
+    crv: algToSecCrv(alg),
+    privateKey: Buffer.from(key, 'hex'),
+  }).toString('pem');
```

Comment:
I'm confused by these functions. Can't we combine them? It seems as though both are functionally the same?

Reply:
It would complicate function logic, which depending on key type will strip or prepend `0x` and uncompress. Besides I am having problem with identifying var, which determines if key is private or public :smile: (alg is going to be renamed to keyType)